### PR TITLE
[PPP-3715] Need to encode inputs to Javascript Rule

### PR DESCRIPTION
**Pull request by Advent51**
- Encoding was added
- Test was added

### CODE REVIEW

**Comment by Advent51** [comment]
@mbatchelor @mchen-len-son please review.
Marc, I couldn't find any way to verify this issue except writing this simple unit test. Do you have any idea how to do that?
I have tried to verify it through writing an `xaction`, but I didn't reach a success.

**Comment by Advent51** [comment]
@mbatchelor please review

**Comment by dkincade** [comment]
Let's get @mbatchelor to review this since it involves encoding...

File: extensions/src/main/java/org/pentaho/platform/plugin/action/javascript/JavascriptRule.java [hunk 1]
```diff
@@ -266,6 +267,7 @@ protected Object executeScript( final ScriptableObject scriptable, final Scripta
         results.setResultSet( (IPentahoResultSet) inputValue );
         wrapper = Context.javaToJS( inputValue, results );
       } else {
+        inputValue = Encode.forHtml( Context.toString( inputValue ) );
```
Comment (mbatchelor) [inline extensions/src/main/java/org/pentaho/platform/plugin/action/javascript/JavascriptRule.java]: I think encoding for html could cause issues here. Suggest killing the encode here as I couldn't demonstrate any ability to subvert the Javascript system based on how we bind the variables. Suggest doing the encode in the success message and calling it a day.

**Comment by Advent51** [comment]
@mbatchelor please review

**Pull request merged**